File: ex/problem_4.ex
# $ elixir -r problem_4.ex -e 'Problem4.main'

defmodule Problem4 do
  def products_of_all_3_digit_numbers() do
    for x <- 100..999, y <- 100..999, do: x * y
  end

  def is_palindrome(n) do
    s = Integer.to_string(n)
    s == String.reverse(s)
  end

  def main do 
    IO.puts (products_of_all_3_digit_numbers() |> Enum.filter(&(is_palindrome/1)) |> Enum.max)
  end
end

File: fs/problem_4.fs
let productsOfTwo3DigitNumbers = [for x in 100..999 do for y in 100..999 -> x * y]

let reverseString (s:string) = new string(Array.rev (s.ToCharArray()))
let isPalindrome x = 
  let s = x.ToString()
  s = reverseString s

printfn "%d" (productsOfTwo3DigitNumbers |> List.filter isPalindrome |> List.max)